src/Microsoft.Health.Fhir.Shared.Api/Features/Formatters/HttpExtensions.cs

```diff
@@ -37,5 +38,33 @@ public static SummaryType GetSummaryType(this HttpContext context, ILogger logge
 
             return SummaryType.False;
         }
+
+        public static bool GetIsPretty(this HttpContext context, ILogger logger)
+        {
+            var query = context.Request.Query[KnownQueryParameterNames.Pretty].FirstOrDefault();
+
+            if (!string.IsNullOrWhiteSpace(query))
+            {
+                try
+                {
+                    if (!bool.TryParse(query, out bool isPretty))
+                    {
+                        // Assume no pretty formatting if parameter can't be parsed.
+                        isPretty = default;
+                    }
+
+                    logger.LogDebug("Changing response pretty formatting to '{0}'", isPretty);
+
+                    return isPretty;
+                }
+                catch (Exception ex)
```

Comment:
I think we can forgo the exception logging here. It should be getting logged even without this.

Reply:
Good to know. Removed in d40c88d.